applications/nrf5340_audio/src/bluetooth/bt_management/bt_mgmt_scan.c

```diff
@@ -7,141 +7,30 @@
 #include "bt_mgmt.h"
 
 #include <zephyr/bluetooth/bluetooth.h>
-#include <zephyr/bluetooth/conn.h>
-#include <zephyr/bluetooth/hci.h>
 
-#include "ble_hci_vsc.h"
+#include "bt_mgmt_scan_for_broadcast_internal.h"
+#include "bt_mgmt_scan_for_conn_internal.h"
 
 #include <zephyr/logging/log.h>
 LOG_MODULE_DECLARE(bt_mgmt);
 
-#define CONNECTION_PARAMETERS                                                                      \
-	BT_LE_CONN_PARAM(CONFIG_BLE_ACL_CONN_INTERVAL, CONFIG_BLE_ACL_CONN_INTERVAL,               \
-			 CONFIG_BLE_ACL_SLAVE_LATENCY, CONFIG_BLE_ACL_SUP_TIMEOUT)
+static char srch_name[BLE_SEARCH_NAME_MAX_LEN];
 
-static uint8_t bonded_num;
-
-static void bond_check(const struct bt_bond_info *info, void *user_data)
-{
-	char addr_buf[BT_ADDR_LE_STR_LEN];
-
-	bt_addr_le_to_str(&info->addr, addr_buf, BT_ADDR_LE_STR_LEN);
-
-	LOG_DBG("Stored bonding found: %s", addr_buf);
-	bonded_num++;
-}
-
-static void bond_connect(const struct bt_bond_info *info, void *user_data)
+int bt_mgmt_scan_start(uint16_t scan_intvl, uint16_t scan_win, enum bt_mgmt_scan_type type,
+		       char const *const name)
 {
 	int ret;
-	const bt_addr_le_t *adv_addr = user_data;
-	struct bt_conn *conn;
-
-	if (!bt_addr_le_cmp(&info->addr, adv_addr)) {
-		LOG_DBG("Found bonded device");
-
-		ret = bt_le_scan_stop();
-		if (ret) {
-			LOG_WRN("Stop scan failed: %d", ret);
-		}
-
-		ret = bt_conn_le_create(adv_addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS,
-					&conn);
-		if (ret) {
-			LOG_WRN("Create ACL connection failed: %d", ret);
-			bt_mgmt_scan_start(0, 0);
-		}
-	}
-}
-
-static int device_found(uint8_t type, const uint8_t *data, uint8_t data_len,
-			const bt_addr_le_t *addr)
-{
-	int ret;
-	struct bt_conn *conn;
-
-	if ((data_len == DEVICE_NAME_PEER_LEN) &&
-	    (strncmp(DEVICE_NAME_PEER, data, DEVICE_NAME_PEER_LEN) == 0)) {
-		LOG_DBG("Device found");
-
-		ret = bt_le_scan_stop();
-		if (ret) {
-			LOG_WRN("Stop scan failed: %d", ret);
-		}
-
-		ret = bt_conn_le_create(addr, BT_CONN_LE_CREATE_CONN, CONNECTION_PARAMETERS, &conn);
-		if (ret) {
-			LOG_ERR("Could not init connection");
-			bt_mgmt_scan_start(0, 0);
-			return ret;
-		}
-
-		return 0;
-	}
-
-	return -ENOENT;
-}
 
-/**
- * @brief  Parse BLE advertisement package
- */
-static void ad_parse(struct net_buf_simple *p_ad, const bt_addr_le_t *addr)
-{
-	while (p_ad->len > 1) {
-		uint8_t len = net_buf_simple_pull_u8(p_ad);
-		uint8_t type;
-
-		/* Check for early termination */
-		if (len == 0) {
-			return;
-		}
-
-		if (len > p_ad->len) {
-			LOG_WRN("AD malformed");
-			return;
-		}
-
-		type = net_buf_simple_pull_u8(p_ad);
-
-		if (device_found(type, p_ad->data, len - 1, addr) == 0) {
-			return;
-		}
+	static int scan_interval = CONFIG_BT_BACKGROUND_SCAN_INTERVAL;
+	static int scan_window = CONFIG_BT_BACKGROUND_SCAN_WINDOW;
```

Comment:
Possibly outside the scope of this PR: Are the value ranges of these Kconfig the same value range of is allowed?